dockers/llm.vdb.service/createvectordb.py

```diff
@@ -165,29 +187,54 @@ def str_to_int(value, name):
             os.makedirs(local_tmp_dir)
 
         # download text docs from S3 bucket + folder (vectordb_s3_input_dir/arg) into this tmp local directory
-        num_files = download_files_from_s3(vectordb_bucket, vectordb_input_arg, local_tmp_dir)
-        print(f"Number of files downloaded is {num_files}, local tmp dir is {local_tmp_dir}")
+        num_files = download_files_from_s3(
+            vectordb_bucket, vectordb_input_arg, local_tmp_dir
+        )
+        print(
+            f"Number of files downloaded is {num_files}, local tmp dir is {local_tmp_dir}"
+        )
 
-        loader = DirectoryLoader(local_tmp_dir, glob="**/*")    
+        loader = DirectoryLoader(local_tmp_dir, glob="**/*")
         documents = loader.load()
-        print(f"Number of documents loaded via DirectoryLoader is {len(documents)}") 
+        print(f"Number of documents loaded via DirectoryLoader is {len(documents)}")
 
         # TODO (improvement for later) Allow users to configure chunk size and overlap values
-        text_splitter = CharacterTextSplitter(chunk_size=embedding_chunk_size, chunk_overlap=embedding_chunk_overlap)
+        text_splitter = CharacterTextSplitter(
+            chunk_size=embedding_chunk_size, chunk_overlap=embedding_chunk_overlap
+        )
         docs = text_splitter.split_documents(documents)
-        
+
         # default model name values has been deprecated since 0.2.16, so we choose a specific model
         embeddings = HuggingFaceEmbeddings(model_name=embedding_model_name)
 
         vectorstore = FAISS.from_documents(docs, embeddings)
+
+    elif vectordb_input_type == "jira-embedding-json":
+        local_tmp_dir = "/tmp/" + vectordb_file
+
+        # create this temp dir if it does not already exist
+        if not os.path.exists(local_tmp_dir):
+            os.makedirs(local_tmp_dir)
+
+        # download text docs from S3 bucket + folder (vectordb_s3_input_dir/arg) into this tmp local directory
+        num_files = download_files_from_s3(
+            vectordb_bucket, vectordb_input_arg, local_tmp_dir
+        )
+        print(
+            f"Number of files downloaded is {num_files}, local tmp dir is {local_tmp_dir}"
+        )
+        vectorstore = create_vectordb(local_tmp_dir, embedding_model_name)
```

Comment:
User configured embedding chunk size and chunk overlap would need to be sent to this method, so it can used by chunk_documents_with_metadata().

Reply:
fixed